skyportal/handlers/api/observation.py

```diff
@@ -0,0 +1,515 @@
+
+  filter  processed_fraction airmass
+0   ztfr                 1.0    None
+1   ztfr                 1.0    None
+2   ztfr                 1.0    None
+3   ztfr                 1.0    None
+4   ztfr                 1.0    None
+     """
+
+    session = Session()
+    try:
+        observations = []
+        for index, row in obstable.iterrows():
+            field_id = int(row["field_id"])
+            field = (
+                session.query(InstrumentField)
+                .filter_by(instrument_id=instrument_id, field_id=field_id)
+                .first()
+            )
+            if field is None:
+                return log(
+                    f"Unable to add observations for instrument {instrument_id}: Missing field {field_id}"
+                )
+
+            observation = (
+                session.query(ExecutedObservation)
+                .filter_by(
+                    instrument_id=instrument_id, observation_id=row["observation_id"]
+                )
+                .first()
+            )
+            if observation is not None:
+                log(
+                    f"Observation {row['observation_id']} for instrument {instrument_id} already exists... continuing."
+                )
+                continue
+
+            observations.append(
+                ExecutedObservation(
+                    instrument_id=instrument_id,
+                    observation_id=row["observation_id"],
+                    instrument_field_id=field.id,
+                    obstime=Time(row["obstime"], format='jd').datetime,
+                    seeing=row["seeing"],
+                    limmag=row["limmag"],
+                    exposure_time=row["exposure_time"],
+                    filt=row["filter"],
+                    processed_fraction=row["processed_fraction"],
+                )
+            )
+        session.add_all(observations)
+        session.commit()
+
+        return log(f"Successfully added observations for instrument {instrument_id}")
+    except Exception as e:
+        return log(f"Unable to add observations for instrument {instrument_id}: {e}")
+    finally:
+        Session.remove()
+
+
+class ObservationHandler(BaseHandler):
+    @permissions(['Upload data'])
+    def post(self):
+        """
+        ---
+        description: Ingest a set of ExecutedObservations
+        tags:
+          - observations
+        requestBody:
+          content:
+            application/json:
+              schema: ObservationHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        telescope_name = data.get('telescopeName')
+        instrument_name = data.get('instrumentName')
+        observation_data = data.get('observationData')
```

Comment:
Should default to an empty dict or else line 162 will error if the user forgets to add this.
```suggestion
        observation_data = data.get('observationData', {})
```